### PR TITLE
Tax item docs draft

### PR DESCRIPTION
**Pull request by hazbo**

### CODE REVIEW

File: carts-and-checkout/carts/README.md [hunk 1]
```diff
@@ -46,6 +46,11 @@ If you do not pass a **`X-MOLTIN-CURRENCY`** header specifying what currency you
                 "amount": 5891,
                 "currency": "USD",
                 "formatted": "$58.91"
+            },
+            "tax": {
+                "amount": 0,
+                "currency": "USD",
+                "formatted": "$0.00"
             }
```
Comment (marcqualie) [inline carts-and-checkout/carts/README.md]: Rogue bracket

File: carts-and-checkout/carts/cart-items/README.md [hunk 2]
```diff
@@ -144,7 +156,10 @@ The `image` object will be empty if there is no main image assigned.
 | `meta.display_price.without_tax.value` | `object` | Tax exclusive totals for this cart item based on the quantity |
 | `meta.display_price.without_tax.value.amount` | `integer` | The raw total price this cart item line \(ex tax\) |
 | `meta.display_price.without_tax.value.currency` | `string` | The currency set for this cart item |
-| `meta.display_price.without_tax.value.formatted` | `string` | The tax inclusive formatted total of this cart item line based on the currency |
+| `meta.display_price.without_tax.value.formatted` | `string` | The tax inclusive formatted total of this cart item line based on the currency | 
+| `meta.display_price.tax.amount` | `integer` | The subtotal of the added tax value |
```
Comment (marcqualie) [inline carts-and-checkout/carts/cart-items/README.md]: This doesn't match the example doc (`meta.display_price.tax.unit.amount`)

File: orders-and-customers/orders/get-all-orders.md [hunk 3]
```diff
@@ -88,6 +88,11 @@ Filter attributes
             "currency": "USD",
             "formatted": "$2175.00"
           }
```
Comment (marcqualie) [inline orders-and-customers/orders/get-all-orders.md]: comma

File: orders-and-customers/orders/get-an-order.md [hunk 4]
```diff
@@ -79,6 +79,11 @@ The Bearer token to grant access to the API
           "currency": "USD",
           "formatted": "$2175.00"
         }
+        "tax": {
```
Comment (marcqualie) [inline orders-and-customers/orders/get-an-order.md:82]: comma

File: orders-and-customers/orders/order-items.md [hunk 5]
```diff
@@ -66,7 +66,19 @@
...
+                            "amount": 40000,
+                            "currency": "USD",
+                            "formatted": "$400.00"
+                        },
+                        "value": {
+                            "amount": 40000,
+                            "currency": "USD",
+                            "formatted": "$400.00"
+                        }
+                    },
```
Comment (marcqualie) [inline orders-and-customers/orders/order-items.md]: json doesn't have trailing commas

File: orders-and-customers/orders/order-items.md [hunk 6]
```diff
@@ -212,6 +228,18 @@ The Bearer token to grant access to the API
               "currency": "GBP",
               "formatted": "£0"
             }
+          },
+          "tax": {
+            "unit": {
```
Comment (marcqualie) [inline orders-and-customers/orders/order-items.md]: What is unit/value?

File: payments/paying-for-an-order/stripe-payments.md [hunk 7]
```diff
@@ -28,6 +28,10 @@ The Bearer token to grant access to the API
 {% endapi-method-headers %}
 
 {% api-method-body-parameters %}
+{% api-method-parameter name="options.idempotency\_key" type="string" required=false %}
```
Comment (marcqualie) [inline payments/paying-for-an-order/stripe-payments.md]: Is this relevant to this PR, or should it be separate?

File: carts-and-checkout/checkout.md [hunk 8]
```diff
@@ -429,6 +429,11 @@ Shipping instructions
                     "currency": "USD",
                     "formatted": "$475.00"
                 }
+                "tax": {
```
Comment (marcqualie) [inline carts-and-checkout/checkout.md:432]: comma

File: carts-and-checkout/carts/cart-items/tax-items/delete-a-tax-item.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+{% endapi-method-request %}
+
+{% api-method-response %}
+{% api-method-response-example httpCode=204 %}
+{% api-method-response-example-description %}
+
+{% endapi-method-response-example-description %}
+
+```javascript
+
```
Comment (marcqualie) [inline carts-and-checkout/carts/cart-items/tax-items/delete-a-tax-item.md:34]: Is there a missing example?

File: carts-and-checkout/carts/cart-items/tax-items/delete-a-tax-item.md [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+{% api-method-headers %}
+{% api-method-parameter name="Authorization" type="string" required=true %}
+The Bearer token to grant access to the API
+{% endapi-method-parameter %}
+{% endapi-method-headers %}
+{% endapi-method-request %}
+
+{% api-method-response %}
+{% api-method-response-example httpCode=204 %}
```
Comment (marcqualie) [inline carts-and-checkout/carts/cart-items/tax-items/delete-a-tax-item.md:28]: We actually return the taxItem, so is this httpCode not invalid?

File: payments/paying-for-an-order/stripe-payments.md (hunk 7 above)
Comment (NullPrice) [reply payments/paying-for-an-order/stripe-payments.md]: I think this has been added in due to a botched rebase, I'll sort it out.

File: carts-and-checkout/carts/cart-items/tax-items/delete-a-tax-item.md (hunk 9 above)
Comment (NullPrice) [reply carts-and-checkout/carts/cart-items/tax-items/delete-a-tax-item.md:34]: That endpoint returns an empty body and a 204. I didn't add that part in but I assume it's just there to match up with the format of other docs.

File: orders-and-customers/orders/order-items.md (hunk 6 above)
Comment (NullPrice) [reply orders-and-customers/orders/order-items.md]: `Unit` is for the tax amount per item where as `value` is the total for that item based on what quantity you have. I'll see if there's a good place for this to be added in as I don't think it's explained anywhere for the other fields.

File: carts-and-checkout/carts/cart-items/tax-items/delete-a-tax-item.md (hunk 10 above)
Comment (NullPrice) [reply carts-and-checkout/carts/cart-items/tax-items/delete-a-tax-item.md:28]: This is how the current implementation handles deletions.

File: carts-and-checkout/carts/cart-items/README.md [hunk 11]
```diff
@@ -144,7 +156,15 @@ The `image` object will be empty if there is no main image assigned.
...
 | `meta.display_price.without_tax.value.amount` | `integer` | The raw total price this cart item line \(ex tax\) |
 | `meta.display_price.without_tax.value.currency` | `string` | The currency set for this cart item |
-| `meta.display_price.without_tax.value.formatted` | `string` | The tax inclusive formatted total of this cart item line based on the currency |
+| `meta.display_price.without_tax.value.formatted` | `string` | The tax inclusive formatted total of this cart item line based on the currency | 
+| `meta.display_price.tax.unit` | `object` | Tax totals |
+| `meta.display_price.tax.unit.amount` | `integer` | The subtotal of the added tax value |
+| `meta.display_price.tax.unit.currency` | `string` | The currency set for the tax |
+| `meta.display_price.tax.unit.formatted` | `string` | The formatted value for the tax subtotal |
+| `meta.display_price.tax.unit` | `object` | Tax totals for this cart item based on the quantity  |
+| `meta.display_price.tax.unit.amount` | `integer` | The subtotal of the added tax value |
```
Comment (marcqualie) [inline carts-and-checkout/carts/cart-items/README.md]: Should this second set have `value` instead of `unit`?

File: carts-and-checkout/carts/cart-items/tax-items/create-a-tax-item.md [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+        "code": "MYTAX01",
+        "rate": 0.2
+    }
+}
+```
+{% endapi-method-response-example %}
+
+{% api-method-response-example httpCode=422 %}
+{% api-method-response-example-description %}
+<<<<<<< HEAD
```
Comment (marcqualie) [inline carts-and-checkout/carts/cart-items/tax-items/create-a-tax-item.md]: Merge conflict 😆

File: orders-and-customers/orders/order-items.md [hunk 13]
```diff
@@ -123,6 +135,10 @@
 | `meta.display_price.without_tax.amount` | `integer` | The raw total of this product \(excludes tax\) |
 | `meta.display_price.without_tax.currency` | `string` | The currency this display price has been formatted for |
 | `meta.display_price.without_tax.formatted` | `string` | The tax exclusive formatted total based on the currency |
+| `meta.display_price.tax` | `object` | Tax totals |
+| `meta.display_price.tax.amount` | `integer` | The subtotal of the added tax value |
+| `meta.display_price.tax.currency` | `string` | The currency set for the tax |
+| `meta.display_price.tax.formatted` | `string` | The formatted value for the tax subtotal |
```
Comment (marcqualie) [inline orders-and-customers/orders/order-items.md]: These probably also need updating to reflect `unit`/`value` based on the example above.

File: orders-and-customers/orders/order-items.md (hunk 6 above)
Comment (marcqualie) [reply orders-and-customers/orders/order-items.md]: Why not `unit` and `total`?

File: carts-and-checkout/carts/cart-items/tax-items/create-a-tax-item.md (hunk 12 above)
Comment (NullPrice) [reply carts-and-checkout/carts/cart-items/tax-items/create-a-tax-item.md]: Disappointed in myself now, I don't know how this didn't flag in my rebase.

Comment (NullPrice) [reply carts-and-checkout/carts/cart-items/tax-items/create-a-tax-item.md]: Oh... I didn't save my change to the file. Doh.

File: carts-and-checkout/carts/cart-items/README.md [hunk 14]
```diff
@@ -144,7 +156,15 @@ The `image` object will be empty if there is no main image assigned.
 | `meta.display_price.without_tax.value` | `object` | Tax exclusive totals for this cart item based on the quantity |
 | `meta.display_price.without_tax.value.amount` | `integer` | The raw total price this cart item line \(ex tax\) |
 | `meta.display_price.without_tax.value.currency` | `string` | The currency set for this cart item |
-| `meta.display_price.without_tax.value.formatted` | `string` | The tax inclusive formatted total of this cart item line based on the currency |
+| `meta.display_price.without_tax.value.formatted` | `string` | The tax inclusive formatted total of this cart item line based on the currency |
```
Comment (marcqualie) [inline carts-and-checkout/carts/cart-items/README.md:159]: This should be unit, value is further down on line 167

File: orders-and-customers/orders/order-items.md [hunk 15]
```diff
@@ -115,14 +127,32 @@
...
-| `meta.display_price.with_tax.formatted` | `string` | The tax inclusive formatted total based on the currency |
-| `meta.display_price.without_tax` | `object` | Tax exclusive prices for this product |
-| `meta.display_price.without_tax.amount` | `integer` | The raw total of this product \(excludes tax\) |
-| `meta.display_price.without_tax.currency` | `string` | The currency this display price has been formatted for |
-| `meta.display_price.without_tax.formatted` | `string` | The tax exclusive formatted total based on the currency |
+| `meta.display_price.with_tax` | `object` | Tax inclusive totals |
+| `meta.display_price.with_tax.unit` | `object` | Tax inclusive totals for a single instance of this order item |
+| `meta.display_price.with_tax.unit.amount` | `integer` | The raw price of a single instance this order item \(inc tax\) |
+| `meta.display_price.with_tax.currency` | `string` | The currency set for this order item |
+| `meta.display_price.with_tax.formatted` | `string` | The tax inclusive formatted total of a single instance of this order item based on the currency |
```
Comment (marcqualie) [inline orders-and-customers/orders/order-items.md]: In this block there is:

```
meta.display_price.with_tax.formatted
meta.display_price.with_tax.unit.formatted
meta.display_price.with_tax.value.formatted
```

Are all 3 required?

File: carts-and-checkout/carts/cart-items/README.md (hunk 14 above)
Comment (NullPrice) [reply carts-and-checkout/carts/cart-items/README.md:159]: 167 is for the `value` object in `tax`. That one pointed out is for `without_tax`.

File: orders-and-customers/orders/order-items.md (hunk 6 above)
Comment (cheeryfella) [reply orders-and-customers/orders/order-items.md]: This is what was initially stipulated way back when. It's a breaking change if we alter this.

File: orders-and-customers/orders/order-items.md (hunk 15 above)
Comment (cheeryfella) [reply orders-and-customers/orders/order-items.md]: tax could be inferred from the other 2 but for completeness and ease for consuming clients we present that information.

**Review by marcqualie — APPROVED** [review]

File: carts-and-checkout/carts/cart-items/tax-items/create-a-tax-item.md [hunk 16]
```diff
@@ -0,0 +1,96 @@
...
+{% endapi-method-parameter %}
+
+{% api-method-parameter name="code" type="string" required=true %}
+A unique tax code for this kind of tax item
+{% endapi-method-parameter %}
+
+{% api-method-parameter name="rate" type="float" required=true %}
+The tax rate percentage represented as a float
+{% endapi-method-parameter %}
+{% endapi-method-body-parameters %}
```
Comment (craigtweedy) [inline carts-and-checkout/carts/cart-items/tax-items/create-a-tax-item.md:40]: `jurisdiction` is missing from the method params?

File: carts-and-checkout/carts/cart-items/tax-items/create-a-tax-item.md [hunk 17]
```diff
@@ -0,0 +1,96 @@
...
+    -H "Authorization: Bearer XXXX" \
+    -d $'{
+        "data": {
+            "type": "tax_item",
+            "name": "Tax Name",
+            "jurisdiction" : "UK",
+            "code": "MYTAX01",
+            "rate": 0.2
+        }
+    }' https://api.moltin.com/v2/carts/:cartID/items/:itemID/taxes
```
Comment (notrab) [inline carts-and-checkout/carts/cart-items/tax-items/create-a-tax-item.md]: This should be after `POST` before `\`
